device-api/src/hwmon.rs

```diff
@@ -107,10 +106,36 @@ struct MetricEntry {
     metric_item: MetricItem,
 }
 
-impl TryFrom<DirEntry> for MetricEntry {
+impl TryFrom<std::fs::DirEntry> for MetricEntry {
     type Error = error::HwmonError;
 
-    fn try_from(value: DirEntry) -> Result<Self, Self::Error> {
+    fn try_from(value: std::fs::DirEntry) -> Result<Self, Self::Error> {
```

Comment:
This is the same as the `try_from` implementation for `TryFrom<tokio::fs::DirEntry>`. Is it available to separate it into a function that takes a string and let each implementation call it?